AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/QtPyScriptCanvasVariableManager.py

```diff
@@ -0,0 +1,119 @@
+"""
+Copyright (c) Contributors to the Open 3D Engine Project.
+For complete copyright and license terms please see the LICENSE at the root of this distribution.
+
+SPDX-License-Identifier: Apache-2.0 OR MIT
+
+Object to house all the Qt Objects and behavior used in testing the script canvas variable manager
+"""
+from editor_python_test_tools.utils import TestHelper as helper
+from PySide2 import QtWidgets, QtCore
+import pyside_utils
+from types import SimpleNamespace
+from consts.scripting import (VARIABLE_MANAGER_QT, VARIABLE_PALETTE_QT, ADD_BUTTON_QT, GRAPH_VARIABLES_QT)
+from consts.general import (WAIT_TIME_3)
+from editor_python_test_tools.utils import Report
+
+
+class Tests():
+    variable_count_expected = ("Variable count matches expected", "Variable count does not match expected")
+
+
+"""
+Basic variable types.
+These are str-str kv pairings so they can be used with SimpleNamespace class easily or as an iterable type.  
+"""
+VARIABLE_TYPES_DICT = {
+    "Boolean": "Boolean",
+    "Color": "Color",
+    "EntityId": "EntityId",
+    "Number": "Number",
+    "String": "String",
+    "Transform": "Transform",
+    "Vector2": "Vector2",
+    "Vector3": "Vector3",
+    "Vector4": "Vector4"
+}
+
+
+class QtPyScriptCanvasVariableManager():
+    """
+    QtPy class for handling the behavior of the script canvas variable manager
+    """
+
+    def __init__(self, sc_editor):
+
+        self.variable_manager = sc_editor.sc_editor.findChild(QtWidgets.QDockWidget, VARIABLE_MANAGER_QT)
+        self.variable_types = self.get_basic_variable_types()
+
+
+    def __validate_new_variable(self, new_variable_type):
+        """
+        function for checking the provided variable type for validity
+        """
+        if type(new_variable_type) is not str:
+            Report.critical_result(["Invalid variable type provided", ""], False)
```

Comment:
consider using assert rather than Report.critical_result since these helpers may out live Report.critical_result and their calling consumer may not be handling Report some day.

Reply:
changed this to an assert